lib/Conversion/TorchToLinalg/TensorConstructors.cpp

```diff
@@ -45,36 +45,69 @@ class ConvertAtenConstantPadNdOp
     auto type = self.getType().cast<RankedTensorType>();
     int64_t rank = type.getRank();
 
-    // Pattern match against the op's original operands, because otherwise we
-    // will get the lowered version of the operands which is harder to pattern
-    // match.
-    SmallVector<int64_t> padInts;
-    if (!matchPattern(op.getPad(), m_TorchListOfConstantInts(padInts)))
-      return rewriter.notifyMatchFailure(
-          op, "only support constant int pad ranges");
-    uint64_t padRank = padInts.size() / 2;
-    if (padRank * 2 != padInts.size())
+    auto primList = op.getPad().getDefiningOp<Torch::PrimListConstructOp>();
+    if (!primList) {
+      return rewriter.notifyMatchFailure(op, "unable to get pad values");
+    }
+
+    SmallVector<Value> padVals(primList.getOperands());
+
+    uint64_t padRank = padVals.size() / 2;
+    if (padRank * 2 != padVals.size())
       return rewriter.notifyMatchFailure(op, "pad range size is not even");
     if (rank < 0 || padRank > (uint64_t)rank)
       return rewriter.notifyMatchFailure(op, "padding exceeds tensor rank");
 
     // Initialize low/high paddings with the dims that should not be padded.
-    SmallVector<int64_t, 4> lowPadding(/*Size=*/rank - padRank, /*Value=*/0);
-    SmallVector<int64_t, 4> highPadding(/*Size=*/rank - padRank, /*Value=*/0);
+    int64_t noPad = rank - padRank;
+    Attribute zero = rewriter.getIndexAttr(0);
+    SmallVector<int64_t, 4> staticLow(noPad, 0);
+    SmallVector<int64_t, 4> staticHigh(noPad, 0);
+    SmallVector<OpFoldResult, 4> lowPad(noPad, zero);
+    SmallVector<OpFoldResult, 4> highPad(noPad, zero);
```

Comment:
nit: what's the convention for including `, 4>` in torch-mlir? I recall some suggestions to drop the number and let the SmallVector template default figure that out on its own